### PR TITLE
Exclude parent elements from caseInsensitiveText search

### LINKED ISSUE #533: exclude parents for By.caseInsensitiveText(TeXt) search
By.caseInsensitiveText(TeXt) search should not include ancestors

**To Reproduce**
When I change context to an element by By.xpath(.//div[@class='uui-popper'])
When I click on element located `By.caseInsensitiveText(DotNet)`

**Expected behavior**
text()=’dotNet’ is equal to 1 so 
Total number of elements found By.xpath is equal to 1
By.caseInsensitiveText(TeXt) search does NOT include ancestors

**Actual**
Total number of elements found By.xpath: .//*[text()[normalize-space(translate(., 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz'))="dotnet"] or @*[normalize-space(translate(., 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz'))="dotnet"] or *[normalize-space(translate(., 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz'))="dotnet"]] is equal to 5

image

2020-05-05_16-16-42_322-Assertion_Failure-chrome-1936x1056

### PR DESCRIPTION
**Pull request by Yauhenda**
Closes #533

### CODE REVIEW

**Comment by valfirst** [comment]
LGTM, however it would be good to add integration tests

File: vividus-tests/src/main/resources/story/integration/FindByTextTests.story [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Meta:
+    @epic vividus-plugin-web-app
+
+Scenario: Verify that caseInsensitiveText search excluding parent elements
+Given I am on a page with the URL '${vividus-test-site-url}/links.html'
+Then number of elements found by `By.caseInsensitiveText(LINK TO AN ELEMENT)` is equal to `1`
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/integration/FindByTextTests.story:6]: By.caseInsensitiveText -> caseInsensitiveText

File: vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/search/AbstractElementSearchAction.java [hunk 2]
```diff
@@ -139,11 +139,16 @@ private boolean isElementVisible(WebElement element, boolean scrolled)
 
     protected static By generateCaseInsensitiveLocator(String text, String... tagNames)
     {
-        StringBuilder locator = new StringBuilder();
+        StringBuilder locator = new StringBuilder(INITIAL_SB_SIZE);
```
Comment (ikalinin1) [inline vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/search/AbstractElementSearchAction.java:142]: https://openjdk.java.net/jeps/280 use concatenation?

File: vividus-tests/src/main/resources/story/integration/FindByTextTests.story [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Meta:
+    @epic vividus-plugin-web-app
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/integration/FindByTextTests.story:2]: add issueId

File: vividus-tests/src/main/resources/story/integration/FindByTextTests.story [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Meta:
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/integration/FindByTextTests.story:1]: Give a story more generic name like ElementSearchTests

File: vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/search/AbstractElementSearchAction.java (hunk 2 above)
Comment (valfirst) [reply vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/search/AbstractElementSearchAction.java:142]: 1) I recommend to remove `INITIAL_SB_SIZE`
2) JEP 280 - I would make it as a separate commit

File: vividus-tests/src/main/resources/story/integration/FindByTextTests.story (hunk 3 above)
Comment (valfirst) [reply vividus-tests/src/main/resources/story/integration/FindByTextTests.story:2]: but at scenario level, not story

**Pull request merged**